src/algopy_testing/models/contract.py

```diff
@@ -42,24 +53,84 @@ def __init_subclass__(
         *,
         name: str | None = None,
         scratch_slots: (
-            algopy.UInt64 | tuple[int | algopy.UInt64, ...] | list[int | algopy.UInt64] | None
+            algopy.urange | tuple[int | algopy.urange, ...] | list[int | algopy.urange] | None
         ) = None,
         state_totals: StateTotals | None = None,
     ):
         cls._name = name or cls.__name__
         cls._scratch_slots = scratch_slots
         cls._state_totals = state_totals
 
+    def _get_local_states(self) -> dict[bytes, algopy.LocalState[Any]]:
+        import algopy
+
+        local_states = {
+            attribute._key.value: attribute
+            for _, attribute in vars(self).items()
+            if isinstance(attribute, algopy.LocalState)
+        }
+
+        return local_states
+
+    def _get_global_states(self) -> dict[bytes, algopy.GlobalState[Any]]:
+        import algopy
+
+        global_states = {}
+        for key, attribute in vars(self).items():
+            if not isinstance(attribute, algopy.LocalState) and not callable(attribute):
```

Comment:
Also need to check here that it is not an instance of `Box`, `BoxRef` or `BoxMap`

Reply:
Resolved, feel free to reopen if you think something else is missing